### PR TITLE
Fix spending htlc output in api server

### PR DESCRIPTION
**Pull request by azarovh**
Was causing `MissingOutputOrSpent` because Htlc outputs were ignored and not added to the db.

Also enabled htlc for api server's simulation.

### CODE REVIEW

**Review by ImplOfAnImpl — APPROVED** [review]

**Review by OBorce — COMMENTED** [review]
I think the address amounts in the db are not being updated when creating and spending the htlc

**Comment by azarovh** [comment]
> I think the address amounts in the db are not being updated when creating and spending the htlc

Yes, the idea is that amount from htlc is not accounted in the balances, because it's not actually yours unless atomic swap is complete as it can be reverted by refund.

File: api-server/scanner-lib/src/blockchain_state/mod.rs [hunk 1]
```diff
@@ -1932,7 +1961,7 @@ fn get_tx_output_destination(txo: &TxOutput) -> Option<&Destination> {
         | TxOutput::Burn(_)
         | TxOutput::DelegateStaking(_, _)
         | TxOutput::DataDeposit(_)
+        | TxOutput::Htlc(_, _)
```
Comment (azarovh) [inline api-server/scanner-lib/src/blockchain_state/mod.rs:1964]: this is wrong, will push a fix shortly

Comment (azarovh) [reply api-server/scanner-lib/src/blockchain_state/mod.rs]: done

**Review by OBorce — APPROVED** [review]
looks ok

**Pull request merged**